code/modules/reagents/chemistry/reagents/pyrotechnic.dm

```diff
@@ -30,6 +30,12 @@
 		C.adjustPlasma(10)
 	..()
 
+/datum/reagent/plasma/on_plasmaman_life(mob/living/carbon/human/H)
+	H.adjustBruteLoss(-0.5 * REAGENTS_EFFECT_MULTIPLIER)
+	H.adjustFireLoss(-0.5 * REAGENTS_EFFECT_MULTIPLIER)
+	H.adjustPlasma(20)
+	H.reagents.remove_reagent(id, metabolization_rate)
```

Comment:
I agree, and welcome suggestions on how to avoid that without doubling up removal since on_mob_life almost universally has ..() calls which remove the reagent